video/stats/src/videoencoderstats.rs

```diff
@@ -0,0 +1,189 @@
+        if self.time_last_buffers.len() > self.max_buffers_inside {
+            self.max_buffers_inside = self.time_last_buffers.len();
+        }
+        gst::log!(CAT, "Current buffers lenght {}", self.time_last_buffers.len());
+    }
+
+    pub fn buffer_out(&mut self) {
+        if let Some(arrive) = self.time_last_buffers.pop_front() {
+            let diff = arrive.elapsed();
+            self.total_processing_time += diff;
+        } else {
+            panic!("output buffer w/o input");
+        }
+    }
+
+    pub fn avg_processing_time(&self) -> Duration {
+        if self.num_buffers != 0 {
+            self.total_processing_time / self.num_buffers as u32
+        } else {
+            Duration::ZERO
+        }
+    }
+}
+
+impl fmt::Display for VideoEncoderStats {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        if self.framerate.unwrap().denom() != 1 {
+            unimplemented!();
+        }
+
+        writeln!(
+            f,
+            "Encoder: {}",
+            &self.name
+        )?;
+        writeln!(
+            f,
+            "Output size: {} KB",
+            self.num_bytes / 1024, // Convert to KB
+        )?;
+
+        let framerate = self.framerate.unwrap();
+        let total_time_secs = self.num_buffers as f64 / framerate.numer() as f64;
+        let bitrate = if total_time_secs > 0.0 {
+            (self.num_bytes as f64 * 8.0) / total_time_secs
+        } else {
+            0.0
+        };
+        let bitrate_str = bitrate/1024.0; // Convert to kbps
+
+        writeln!(f, "Bitrate: {:.3} kbps", bitrate_str)?;
+
+        let avg_processing_time = self.avg_processing_time().as_millis();
+        writeln!(
+            f,
+            "Processing time: {:.2} ms",
+            avg_processing_time
+        )?;
+
+        let cpu_time = self.threads_utime + self.threads_stime;
+        #[cfg(target_os = "linux")]
+        let cpu_time_seconds = {
+            let ticks_per_second = procfs::ticks_per_second() as u64;
+            cpu_time as f64 / ticks_per_second as f64
+        };
+        writeln!(
+            f,
+            "CPU: {} s",
+            cpu_time_seconds
+        )?;
+
+        let vmaf_score_str = match self.vmaf_score {
+            Some(score) => format!("{:.3}", score),
+            None => "N/A".to_string(),
+        };
+        writeln!(f, "VMAF: {}", vmaf_score_str)?;
+
+        let pre_encode_time = &self.pre_encode_time;
+        let post_encode_time = &self.post_encode_time;
+        let encode_latency = (*post_encode_time as f64 - *pre_encode_time as f64) / 1_000_000.0;
+        writeln!(
+            f,
+            "Encode latency: {:.3} ms",
+            encode_latency
+        )?;
+
+        let current_time = gst::SystemClock::obtain();
+        let latency = (current_time.time().unwrap().nseconds() as f64 - self.input_time as f64) / 1_000_000.0;
+
+        writeln!(
+            f,
+            "End2End latency: {:.3} ms",
+            latency
+        )
```

Comment:
IMHO I would drop this. it is not End2End latency and IMHO it doesn't add any useful information because we can not assume input_queue is after the source.

Also it out of the scope of `video-encoder-stats`.